athenahealth/httpclient.go

```diff
@@ -162,12 +166,6 @@ func (h *HTTPClient) setBaseURL() {
 }
 
 func (h *HTTPClient) request(ctx context.Context, method, path string, body io.Reader, headers http.Header, out interface{}) (*http.Response, error) {
-	if _, ok := ctx.Deadline(); !ok {
```

Comment:
What takes precedence if the httpclient `Timeout` is set *and* a context with timeout is passed on the individual request, assume the latter? Was poking around the [source](https://cs.opensource.google/go/go/+/refs/heads/master:src/net/http/client.go;drc=7fed33815cf57bf8d6b6ddfbd2ce0f5d8180b4f6;l=193?q=http%20client&ss=go%2Fgo) but didn't see references to the context timeouts, just the client's timeout

Reply:
internally, `http` will respect the context timeout if it exists. Whatever is shorter takes 'precedence'.

Reply:
Interesting - so then are we losing the ability for callers to set a longer timeout on a per-request basis here? Looks like before the default was only applied if no context timeout was set so you could have made a call with a 20-second context timeout for a single request, but now looks like the client timeout would act as a ceiling and override that down to 15? Maybe we want that but if we do perhaps we should rename the default timeout to indicate that behavior. I might also be reading that change wrong though.

Reply:
Ah true this change does break callers that are currently extending the timeout by providing custom context timeouts (i.e. ourselves for image uploads)
A better option would be to allow callers to change the default timeout which we can do through an option at client creation. I'll get that change in.
Nice catch!